realm/realm-jni/build.gradle

```diff
@@ -163,11 +182,40 @@ task downloadCore(group: 'build setup', description: 'Download the latest versio
     }
 }
 
+task compileCore(group: 'build setup', description: 'Compile the core library from source code') {
+    // Build the library from core source code
+    doFirst {
+        if (!coreSourcePath) {
+            throw new GradleException('The coreSourcePath is not set.')
+        }
+        exec {
```

Comment:
And i suppose with `exec` block, the return code will be somehow caught by the `doFirst`?